src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Azure/Functions/AzureFunctionsCommon.cs

```diff
@@ -0,0 +1,138 @@
+                        triggerType = "Automatic";
+
+                        if (bindingSourceFullName.Contains("Timer"))
+                        {
+                            // ex: Microsoft.Azure.WebJobs.Host.Triggers.TriggerBindingSource`1[[Microsoft.Azure.WebJobs.TimerInfo, Microsoft.Azure.WebJobs.Extensions, Version=4.0.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]]
+                            triggerType = "Timer";
+                        }
+                        else if (bindingSourceFullName.Contains("ServiceBus"))
+                        {
+                            // ex: Microsoft.Azure.WebJobs.Host.Triggers.TriggerBindingSource`1[[Microsoft.Azure.WebJobs.ServiceBus.ServiceBusTriggerInput, Microsoft.Azure.WebJobs.ServiceBus, Version=4.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]]
+                            triggerType = "ServiceBus";
+                        }
+                        else if (bindingSourceFullName.Contains("Blob"))
+                        {
+                            // ex: Microsoft.Azure.WebJobs.Host.Triggers.TriggerBindingSource`1[[Microsoft.Azure.Storage.Blob.ICloudBlob, Microsoft.Azure.Storage.Blob, Version=11.1.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]]
+                            triggerType = "Blob";
+                        }
+                        else if (bindingSourceFullName.Contains("Microsoft.Azure.DocumentDB.Core"))
+                        {
+                            // ex: Microsoft.Azure.WebJobs.Host.Triggers.TriggerBindingSource`1[[System.Collections.Generic.IReadOnlyList`1[[Microsoft.Azure.Documents.Document, Microsoft.Azure.DocumentDB.Core, Version=2.13.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]], System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]
+                            triggerType = "Cosmos";
+                        }
+
+                        break;
+                    case AzureFunctionsExecutionReason.Dashboard:
+                        triggerType = "Dashboard";
+                        break;
+                }
+
+                var functionName = instanceParam.FunctionDescriptor.ShortName;
+
+                var tags = new AzureFunctionsTags
+                {
+                    TriggerType = triggerType,
+                    ShortName = functionName,
+                    FullName = instanceParam.FunctionDescriptor.FullName,
+                    BindingSource = bindingSourceType.FullName
+                };
+
+                if (tracer.ActiveScope == null)
+                {
+                    // This is the root scope
+                    tags.SetAnalyticsSampleRate(IntegrationId, tracer.Settings, enabledWithGlobalSetting: false);
+                    scope = tracer.StartActiveWithTags(OperationName, tags: tags);
+                }
+                else
+                {
+                    scope = tracer.StartActive(OperationName);
```

Comment:
Should the tags be set on this span too, or just on the root?

Reply:
Intentionally decided to do just the root, but I am open to argument

Reply:
I'm okay with either. If we start with just tagging the root span, we can always go back later and add tags to inner spans later